File: sdk/SendCraft.Sdk/SendCraftException.cs
using System.Text.Json.Nodes;

namespace SendCraft.Sdk;

public sealed class SendCraftException : Exception
{
    public int StatusCode { get; }
    public JsonNode? Payload { get; }

    public SendCraftException(string message, int statusCode, JsonNode? payload = null)
        : base(message)
    {
        StatusCode = statusCode;
        Payload = payload;
    }
}

File: sdk/sendcraft-java-sdk/src/main/java/com/sendcraft/sdk/SendCraftException.java
package com.sendcraft.sdk;

import com.fasterxml.jackson.databind.JsonNode;

public class SendCraftException extends RuntimeException {
    private final int statusCode;
    private final JsonNode payload;

    public SendCraftException(String message, int statusCode, JsonNode payload) {
        super(message);
        this.statusCode = statusCode;
        this.payload = payload;
    }

    public int getStatusCode() {
        return statusCode;
    }

    public JsonNode getPayload() {
        return payload;
    }
}